core/io/linux.rs

```diff
@@ -58,11 +80,25 @@ impl File for LinuxFile {
             let mut buf = c.buf_mut();
             let len = buf.len();
             let buf = buf.as_mut_ptr();
-            let ptr = Rc::into_raw(c.clone());
-            io_uring::opcode::Read::new(fd, buf, len as u32)
+            let ptr: *const Completion = Rc::into_raw(c.clone());
+            //TODO: Check where to instantiate Probe
+            let probe = io_uring::Probe::new();
+            if probe.is_supported(io_uring::opcode::Read::CODE) {
```

Comment:
A probe is a system call so let's move it out of the read path.

Reply:
yes I will do.

Reply:
Should this be done on compile time ideally? I don't think a simple boolean `if .. else..` will impose too much overhead due to branch prediction but an open question

Reply:
meh, I think I agree on switching to defaulting to readv. We will see penalties in profiles if we are wrong anyways and it won't be hard to change.

Reply:
I will see options, first I want to have the system working on my end which I am not yet there.

Thanks for the feedback!

Reply:
program startup seems like the correct place to Probe. I would check at startup and set a static attribute on the IO.

Reply:
> @pereman2 I actually don't think this can be compile time because -- at least in theory -- you want to be able to run the same Linux binary across different kernel versions. There seem to be pretty old versions of io_uring still out there that don't even support probing properly...
> 
> 
> 
> _If_ we probe, we should do that at program startup, and then stick to a specific I/O method.

Truee then yeah, startup